cpp/include/cudf/detail/contiguous_split.hpp

```diff
@@ -105,9 +105,23 @@ class metadata_builder {
    */
   std::vector<uint8_t> build() const;
 
+  /**
+   * @brief Clear the internal buffer containing all added metadata.
+   */
+  void clear();
```

Comment:
we never reuse metadata, it's return and throw away. I don't see a lot of value in a `clear()` method.

Reply:
We don't reuse the data, but we need to reuse the temp buffer. Previously we allocate a temp buffer for each returned partition then discard that buffer away, which is kind of wasteful. Now we can reuse the buffer to store metadata. The `clear` method is just empty out the current data to reuse for the next column. The buffer is still there, and can save us some time.

Reply:
ah ok I think I see how this works. You are creating 1 builder, then clearing each time between contig tables, that would work.